### PR TITLE
Add support for pointer literals in metal

### PR DESCRIPTION
**Pull request by jarcherNV**
Add support for kIROp_PtrLit types in metal and add a test for both null and non-null pointers.

### CODE REVIEW

File: tests/metal/pointer-literals.slang [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    // Test 3: Pointer literal in conditional
+    bool useNull = true;
+    void* ptr2;
+    if (useNull)
+    {
+        ptr2 = nullptr;
+    }
+    else
+    {
+        ptr2 = (void*)0x1234;
```
Comment (jkwak-work) [inline tests/metal/pointer-literals.slang]: I am not sure if we want to have this kind of explicit address setting in our test case.

**Review by jkwak-work — COMMENTED** [review]
Looks good to me

File: tests/metal/pointer-literals.slang (hunk 1 above)
Comment (jarcherNV) [reply tests/metal/pointer-literals.slang]: That's fair. I have updated it.

**Review by jkwak-work — APPROVED** [review]
Looks good to me.
